components/Cast/Upvote.tsx

```diff
@@ -6,12 +6,17 @@ import { IoIosArrowDropup, IoIosArrowDropupCircle } from 'react-icons/io';
 
 const Upvote = ({ cast }: { cast: Cast }) => {
   const { signer } = useNeynarProvider();
-  const [upvoted, setUpvoted] = useState<boolean>(
-    signer && cast.reactions.likes.some((like: any) => like?.fid === signer?.fid),
-  );
+  const [upvoted, setUpvoted] = useState<boolean>(() => {
+    if (signer && cast.reactions.likes.some((like: any) => like?.fid === signer?.fid)) {
+      return true;
+    }
+    return false;
+  });
```

Comment:
why did you make this change?

more code for exact same result right?

I think we should remove your change and keep code as is

Is there anything I'm missing?